File: Java/MiddleCharacter.java

public class MiddleCharacter {

    static String get_middle(String sample) {
        return sample.length() % 2 == 0 ? sample.substring((sample.length() / 2 - 1), sample.length() / 2) : sample.substring(sample.length() / 2, 1);
    }
    
    public static void main(String[] args) {
        String[] samples = {"Santa", "Hitler", "Germany", "Michael", "Manemjeff"};

        for (String elements : samples)
            System.out.println(elements + " = " + get_middle(elements));
    }
}

File: C++/MiddleCharacter.cpp
#include <iostream>

std::string get_middle(std::string sample)
{
    return sample.length() % 2 == 0 ? sample.substr(sample.length() / 2 - 1, 2) : sample.substr(sample.length() / 2, 1); // even
}

int main()
{
    std::string words[] = {"Santa", "Floccinaucinihilipilification", "Github", "Michael", "supercalifragilisticexpialidocious"};

    for (auto var : words)
        std::cout << var << " = " << get_middle(var) << std::endl;

    return 0;
}
